### ISSUE TITLE
Infix types for ExplicitResultTypes rewrites

### ISSUE THREAD

**Issue by fommil** [scalacenter/scalafix#685]
The type `\/[String, A]` looks much better as `String \/ A`

It would be good if symbolic types with two parameters were added as infix.

**Comment by olafurpg** [comment]
We can maybe use the new `SymbolInformation.annotations` to look for `@showAsInfix` annotations. Should all symbolic types with two parameters be printed as infix or are there exceptions?

**Comment by fommil** [comment]
ensime does it for all symbolic methods, I'm not aware of exceptions

**Comment by olafurpg** [comment]
This is a great idea! A good place to fix this would be here 
https://github.com/scalacenter/scalafix/blob/6d48e0c419a63956274cfcc8e9bb23601c75ccfa/scalafix-core/shared/src/main/scala/scalafix/internal/util/PrettyType.scala#L530

The goal is to use `Type.ApplyInfix` instead of `Type.Apply`

```scala
@ t"A :+: B"
res4: Type.ApplyInfix = Type.ApplyInfix(Type.Name("A"), Type.Name(":+:"), Type.Name("B"))

@ t":+:[A, B]"
res5: Type.Apply = Type.Apply(Type.Name(":+:"), List(Type.Name("A"), Type.Name("B")))
```

If the `qual` is `Type.Name` with a symbolic name (can be checked with `Character.isJavaIdentifierPart` against `name.head`).

A good place to test this would be here 
https://github.com/scalacenter/scalafix/blob/6d48e0c419a63956274cfcc8e9bb23601c75ccfa/scalafix-tests/input/src/main/scala/test/explicitResultTypes/ExplicitResultTypesBase.scala#L54
with expected output in https://github.com/scalacenter/scalafix/blob/6d48e0c419a63956274cfcc8e9bb23601c75ccfa/scalafix-tests/output/src/main/scala/test/explicitResultTypes/ExplicitResultTypesBase.scala#L49

**Comment by fommil** [comment]
sounds like a great spree ticket

**Comment by olafurpg** [comment]
I'll be at a spree next week in Oslo :)

**Comment by oyvindberg** [comment]
I'll see if i can make any progress here :)

**Comment by olafurpg** [comment]
Fixed in #724 🎉
